packages/react-router/__tests__/dom/ssr/components-test.tsx

```diff
@@ -148,125 +148,126 @@ describe("<NavLink />", () => {
   itPrefetchesPageLinks(NavLink);
 });
 
-describe("<RemixServer>", () => {
-  it("handles empty default export objects from the compiler", async () => {
-    let staticHandlerContext = await createStaticHandler([{ path: "/" }]).query(
-      new Request("http://localhost/")
-    );
-    invariant(
-      !(staticHandlerContext instanceof Response),
-      "Expected a context"
-    );
+// TODO: figure out why this test exists
```

Comment:
I'm not sure why these exist but the underlying issue causing them to fail should probably be fixed.

Reply:
These tests were from an issue a while back where IIRC the lack of a default export was being converted to `export default {}` in the compiler and that was causing us to think it had a component export and eventually call `createElement({})` - so we strengthened our check for a component to make sure it was an actual react component.  I don't specifically recall if they were `esbuild` only or also a `vite` thing though.

I got the tests passing and pushed it up

https://github.com/remix-run/remix/issues/7593#issuecomment-1770964626
https://github.com/remix-run/remix/blob/main/packages/remix-react/routes.tsx#L614
https://github.com/remix-run/remix/pull/7745